frontend/src/utils/helpers.js

```diff
@@ -0,0 +1,18 @@
+const nFormatterLookup = [
+  { value: 1, symbol: '' },
+  { value: 1e3, symbol: 'K' },
+  { value: 1e6, symbol: 'M' },
+  { value: 1e9, symbol: 'G' },
+  { value: 1e12, symbol: 'T' },
+  { value: 1e15, symbol: 'P' },
+  { value: 1e18, symbol: 'E' },
+]
+
+export const nFormatter = (num, fractions = 1) => {
+  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/
+  const item = nFormatterLookup
+    .slice()
+    .reverse()
+    .find(item => num >= item.value)
+  return item ? (num / item.value).toFixed(fractions).replace(rx, '$1') + item.symbol : '0'
+}
```

Comment:
I think this could be replaced with react-intl's formatNumber. The following returns the same values as nFormatter does, up to 1e12 at least:
```js
import { useIntl } from 'react-intl'
const intl = useIntl()
intl.formatNumber(t, {
  notation: 'compact',
  compactDisplay: 'short'
})
```

It would support localization and all of the number formats we might specify in the layout JSON.

There would be a straightforward rule for using it too:
1. If the axis's `format` isn't `"number"`, display the raw value and stop here. 
2. Start with `{notation: 'compact', compactDisplay: 'short'}`.
3. Merge the `numberFormat` object into that if it exists in the metric specification. 
4. Drop `minimumFractionDigits`, `minimumIntegerDigits`, and `minimumSignificantDigits`.
5. Pass those options into `intl.formatNumber`.